### PR TITLE
Feed merge

### PR DESCRIPTION
**Pull request by danilkolikov**
+ Posts caching
+ Now loadPosts returns observable of posts to show on screen

### CODE REVIEW

File: app/src/main/java/ly/loud/loudly/util/NetworkUtils.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import android.support.annotation.NonNull;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import ly.loud.loudly.base.single.SinglePost;
+
+public class NetworkUtils {
+    public static class DividedList {
+        public final List<SinglePost> cached;
```
Comment (danilkolikov) [inline app/src/main/java/ly/loud/loudly/util/NetworkUtils.java]: Nullability fixed

File: app/src/main/java/ly/loud/loudly/application/models/PostLoadModel.java [hunk 2]
```diff
@@ -27,9 +38,54 @@ public PostLoadModel(@NonNull Loudly loudlyApplication, @NonNull CoreModel coreM
...
+                        .filter(single -> {
+                            SinglePost instance = loudlyPost
+                                    .getSingleNetworkInstance(single.getNetwork());
+                            return instance != null && instance.getLink().equals(single.getLink());
+                        }) // And set this posts as instances
+                        .reduce(loudlyPost, (lPost, sPost) -> lPost
+                                .setSingleNetworkInstance(sPost.getNetwork(), sPost));
+            }
+            return post;
+        }).collect(ToSolidList.toSolidList());
```
Comment (ZeRoGerc) [inline app/src/main/java/ly/loud/loudly/application/models/PostLoadModel.java]: static import

File: app/src/main/java/ly/loud/loudly/application/models/PostLoadModel.java [hunk 3]
```diff
@@ -27,9 +38,54 @@ public PostLoadModel(@NonNull Loudly loudlyApplication, @NonNull CoreModel coreM
...
+     * @return Observable, containing lists of post to show. First list in stream contains
+     * posts from DB, next contains previous list merged with posts from some new network
+     */
     @CheckResult
     @NonNull
-    public Observable<SolidList<? extends PlainPost>> loadPosts(@NonNull TimeInterval interval) {
-        return coreModel.getConnectedNetworksModels().flatMap(n -> n.loadPosts(interval));
+    public Observable<SolidList<PlainPost>> loadPosts(@NonNull TimeInterval interval) {
+        return Observable
+                .fromCallable(() -> ListUtils.asSolidList(DatabaseUtils.loadPosts(interval)))
```
Comment (ZeRoGerc) [inline app/src/main/java/ly/loud/loudly/application/models/PostLoadModel.java]: static import

**Comment by ZeRoGerc** [comment]
ok with issues

**Pull request merged**